src/components/UserProfile/BlueSquares/BlueSquare.jsx

```diff
@@ -3,77 +3,151 @@ import hasPermission from 'utils/permissions';
+//                   key={index}
+//                   role="button"
+//                   id="wrapper"
+//                   data-testid="blueSquare"
+//                   className="blueSquareButton"
+//                   onClick={() => {
+//                     if (!blueSquare._id) {
+//                       handleBlueSquare(isInfringementAuthorizer, 'message', 'none');
+//                     } else if (canEditInfringements || canDeleteInfringements) {
+//                       handleBlueSquare(
+//                         canEditInfringements || canDeleteInfringements,
+//                         'modBlueSquare',
+//                         blueSquare._id,
+//                       );
+//                     // } else if(canEditInfringements) {
+//                     //   handleBlueSquare(
+//                     //     canEditInfringements,
+//                     //     'editBlueSquare',
+//                     //     blueSquare._id,
+//                     //   );
+//                     // } else if (canDeleteInfringements){
+//                     //   handleBlueSquare(
+//                     //     canDeleteInfringements,
+//                     //     'deleteBlueSquare',
+//                     //     blueSquare._id,
+//                     //   );
+
+//                     }else{
+//                       handleBlueSquare(
+//                         true,
+//                         'viewBlueSquare',
+//                         blueSquare._id,
+//                       );
+//                     }
+//                   }}
+//                 >
+//                   <div className="report" data-testid="report">
+//                     <div className="title">{formatDate(blueSquare.date)}</div>
+//                     {blueSquare.description !== undefined && (
+//                       <div className="summary">
+//                         {blueSquare.createdDate !== undefined && blueSquare.createdDate !== null
+//                           ? `${formatCreatedDate(blueSquare.createdDate)}: ${
+//                               blueSquare.description
+//                             }`
+//                           : blueSquare.description}
+//                       </div>
+//                     )}
+//                   </div>
+//                 </div>
+//               ))
+//           : <div>No blue squares.</div>}
+//         {canAddInfringements && (
+//           <div
+//             onClick={() => {
+//               handleBlueSquare(true, 'addBlueSquare', '');
+//             }}
+//             className="blueSquareButton"
+//             color="primary"
+//             data-testid="addBlueSquare"
+//           >
+//             +
+//           </div>
+//         )}
+//       </div>
+//     </div>
+//   );
+// };
+
+
+/*
+Consolidated destructuring for props.
+Simplified handleOnClick by reducing repeated conditions.
+Moved formatting logic inside JSX.
+Removed redundant conditions and comments for cleaner code.
+*/
+
```

Comment:
```suggestion
```
Same as the backend. Deleting this old code instead of just commenting it out and duplicating it reduces clutter from old code. The comment should be in the commit/PR description as it doesn't make sense to keep after the PR.